frontend/awx/common/PageFormMultiSelectAwxResource.tsx

```diff
@@ -0,0 +1,172 @@
+        };
+      } catch (error) {
+        return {
+          remaining: 0,
+          options: [],
+          next: 0,
+        };
+      }
+    },
+    [props.url, props.queryParams]
+  );
+
+  const [_, setDialog] = usePageDialog();
+  const { setValue } = useFormContext<FormData>();
+  const value = useWatch<FormData>({ name: props.name }) as number[];
+  const openSelectDialog = useCallback(
+    (onSelect: (resource: Resource[]) => void) => {
+      setDialog(
+        <SelectResource<Resource>
+          title={props.label}
+          url={props.url}
+          onSelect={onSelect}
+          toolbarFilters={props.toolbarFilters}
+          tableColumns={props.tableColumns}
+          queryParams={props.queryParams}
+          defaultSelection={
+            value && Array.isArray(value)
+              ? value.map((item: number) => {
+                  return { id: item };
+                })
+              : []
+          }
+        />
+      );
+    },
+    [
+      props.label,
+      props.tableColumns,
+      props.toolbarFilters,
+      props.url,
+      setDialog,
+      value,
+      props.queryParams,
+    ]
+  );
+
+  const queryLabel = useCallback(
+    (value: Value) => <AwxAsyncName url={props.url} id={value as unknown as number} />,
+    [props.url]
+  );
+
+  return (
+    <PageFormAsyncMultiSelect<FormData, Name>
+      id={id}
+      name={props.name}
+      label={props.label}
+      queryOptions={queryOptions}
+      placeholder={props.placeholder}
+      queryPlaceholder={props.queryPlaceholder}
+      queryErrorText={props.queryErrorText}
+      isRequired={props.isRequired}
+      isDisabled={props.isDisabled}
+      helperText={props.helperText}
+      labelHelp={props.labelHelp}
+      onBrowse={() =>
+        openSelectDialog((resources: Resource[]) => {
+          if (resources) {
+            setValue(props.name, resources.map((res) => res.id) as PathValue<FormData, Name>);
+          }
+        })
+      }
+      queryLabel={queryLabel}
+    />
+  );
+}
+
+function SelectResource<
+  Resource extends { id: number; name: string; description?: string | null | undefined },
+>(props: {
+  title: string;
+  url: string;
+  onSelect: (resource: Resource[]) => void;
+  defaultSelection?: { id: number }[];
+  toolbarFilters?: IToolbarFilter[];
+  tableColumns: ITableColumn<Resource>[];
+  queryParams?: QueryParams;
+}) {
+  const view = useAwxView<Resource>({
+    url: props.url,
+    toolbarFilters: props.toolbarFilters,
+    tableColumns: props.tableColumns,
+    disableQueryString: true,
+    defaultSelection: props.defaultSelection as Resource[],
+    queryParams: props.queryParams,
+  });
+  return (
+    <MultiSelectDialog<Resource>
+      title={props.title}
+      onSelect={props.onSelect}
+      toolbarFilters={props.toolbarFilters ?? []}
+      tableColumns={props.tableColumns}
+      view={view}
+    />
+  );
+}
+
+export function constructQueryParams(queryParams: QueryParams) {
```

Comment:
Do we want to put this function in a separate file maybe under utils?

Reply:
I will look if there is any usage of this functions in fw.